src/main/java/org/sonarsource/scanner/cli/Cli.java

```diff
@@ -90,7 +95,12 @@ private int processNextArg(String[] args, int pos) {
       return processProp(args, pos);
 
     } else if ("--embedded".equals(arg)) {
+      logger.info("Option --embedded is deprecated and will be removed in a future release.");
+      embedded = true;
+
+    } else if (arg.startsWith("--from")) {
       embedded = true;
+      invokedFrom = arg.substring("--from=".length());
```

Comment:
Should we check that the from argument actually has a value before assuming it does?
And that the value is in the format the Cli expects?
Now the Cli will default to a common 'from' if the `from` argument is not of the expected format `[name]/[version]`.

Reply:
This check is done in the ScannerFactory directly, we both check for empty / contains a forward slash before actually trying to parse it, otherwise it falls back to the default value of ScannerCLI. 

I was wondering if we should keep a shortlist of possible Scanner Names (at least) in the Scanner Engine so that no one can put garbage on this one. WDYT ?

Reply:
> This check is done in the ScannerFactory directly, we both check for empty

This will check the string value after it is 'extracted" from the `--from` argument. What I mean is, should be check that the `from` argument is not just `--from`? If only `--from` is provided then you will get an index out of bounds exception when trying to get a substring that should start after the original string has already ended.

As for the scanner naming, using something like a coded list or enum from the names limit the extendibility of the parameter. And by this I mean that a code update would be needed in this repo if another wants to use a different name in `from`.
Maybe the best solution would be to allow other values but monitor the behaviour.

Reply:
Indeed for the from check, i'll make the change. And agreed for the naming, let's see how it goes and take decisions after if needed. Thanks !

Reply:
👍